checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from typing import Any, Dict
+
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.arm.base_resource_check import BaseResourceCheck
+
+
+class AzureDefenderOnKeyVaults(BaseResourceCheck):
+    def __init__(self) -> None:
+        name = "Ensure that Azure Defender is set to On for Key Vault"
+        id = "CKV_AZURE_87"
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get('properties', {})
+        pricing_tier = properties.get('pricingTier')
+        return (
```

Comment:
Probably we should also check the name equal to `KeyVaults`
Am I right @tsmithv11?

Reply:
Yeah, correct